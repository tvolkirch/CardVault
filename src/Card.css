.flex-card-wrapper
{
    display: flex;
    flex-direction: column;
    width: 220px;
    border: 2px solid #666;
	-moz-border-radius: 5px;
	-webkit-border-radius: 5px;
	border-radius: 5px;
	-moz-box-shadow: 5px 5px 5px #ccc;
	-webkit-box-shadow: 5px 5px 5px #ccc;
	box-shadow: 5px 5px 5px #ccc;
	background-color: #111;
	padding: 1rem;
	margin: 1rem;
}

.card-content
{
    display: flex;
    flex-direction: column;
    width: 95%;
    color: #ccc;
}

.card-content .card-name
{
    font-size: 1.25rem;
    font-weight: bold;
    text-align: center;
    margin-bottom: 0.5rem;
}

.card-image
{
   height: 300px;
   margin-left: auto;
   margin-right: auto;
   margin-bottom: 1rem;
}

.flex-card-row
{
    display: flex;
}

.flex-card-row span
{
    margin-bottom: 0.5rem;
}

.flex-card-row .left-column
{
    font-weight: bold;
    margin-right: 0.5rem;
    width: 20%;
    text-align: right;
}

.card-rules
{
    margin-top: 0;
    margin-bottom: 0.25rem;
    font-weight: bold;
}
