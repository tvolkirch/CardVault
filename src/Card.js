import React from "react";

import "./Card.css";

function Card(props)
{    
   return (
      <div className="flex-card-wrapper">
         <div className="card-content" tabIndex="0">
            <img className="card-image" src={props.url}
               alt={"Elder Scrolls Card: " + props.name}/>
         </div>
         <div className="card-content card-text" data={props.index} tabIndex="0">
               <div className="card-name">{props.name}</div>
               <div>
                  <div className="flex-card-row">
                     <span className="left-column">Type:</span>
                     <span className="right-column">{props.type}</span>
                  </div>
                  <div className="flex-card-row">
                     <span className="left-column">Set:</span>
                     <span className="right-column">{props.set}</span>
                  </div>
               </div>
               <div className="card-rules">{props.rules}</div>
         </div>
      </div>
   );
}

export default Card;