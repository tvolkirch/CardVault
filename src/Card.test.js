import React from "react";
import renderer from "react-test-renderer";

import Card from "./Card";

it("Consistently renders a card", () => {

   const card = renderer
      .create(<Card url="#"
                   name="Card Name"
                   type="Card Type"
                    set="Card Set Name"
                  rules="Card Text"/>)
      .toJSON();

   expect(card).toMatchSnapshot();
});

