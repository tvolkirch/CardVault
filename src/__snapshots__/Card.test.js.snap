// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consistently renders a card 1`] = `
<div
  className="flex-card-wrapper"
>
  <div
    className="card-content"
  >
    <img
      alt="Elder Scrolls Card: Card Name"
      className="card-image"
      src="#"
      tabIndex="0"
    />
  </div>
  <div
    className="card-content card-text"
    tabIndex="0"
  >
    <div
      className="card-name"
    >
      Card Name
    </div>
    <div>
      <div
        className="flex-card-row"
      >
        <span
          className="left-column"
        >
          Type:
        </span>
        <span
          className="right-column"
        >
          Card Type
        </span>
      </div>
      <div
        className="flex-card-row"
      >
        <span
          className="left-column"
        >
          Set:
        </span>
        <span
          className="right-column"
        >
          Card Set Name
        </span>
      </div>
    </div>
    <div
      className="card-rules"
    >
      Card Text
    </div>
  </div>
</div>
`;
